src/goto-instrument/contracts/havoc_assigns_clause_targets.h

```diff
@@ -49,17 +53,61 @@ class message_handlert;
 /// of their addresses, and havoc them in a second time.
 /// Snapshotting and havocking are both guarded by the validity of the target.
 ///
-void havoc_assigns_clause_targets(
-  const irep_idt &replaced_function_id,
-  const std::vector<exprt> &targets,
-  goto_programt &dest,
-  // context parameters
-  const source_locationt &source_location,
-  const irep_idt &mode,
-  namespacet &ns,
-  symbol_tablet &st,
-  message_handlert &message_handler);
+class havoc_assigns_clause_targetst : public instrument_spec_assignst
+{
+public:
+  havoc_assigns_clause_targetst(
+    const irep_idt &_function_id,
+    const std::vector<exprt> &_targets,
+    const goto_functionst &_functions,
+    const source_locationt &_source_location,
+    namespacet &_ns,
+    symbol_tablet &_st,
```

Comment:
As said in another comment: don't take both a `namespacet` and a `symbol_tablet` (and essentially never take a non-const `namespacet`).